app/src/main/java/io/homeassistant/companion/android/sensors/SensorDetailFragment.kt

```diff
@@ -72,6 +72,32 @@ class SensorDetailFragment(
                 return@setOnPreferenceChangeListener true
             }
         }
+        findPreference<Preference>("description")?.let {
+            var sensorDescription: String = when (sensorId) {
+                "audio_sensor" -> resources.getString(R.string.audio_sensor)
+                "battery_level" -> resources.getString(R.string.battery_level)
+                "battery_state" -> resources.getString(R.string.battery_state)
+                "bluetooth_connection" -> resources.getString(R.string.bluetooth_connection)
+                "detected_activity" -> resources.getString(R.string.detected_activity)
+                "dnd_sensor" -> resources.getString(R.string.dnd_sensor)
+                "geocoded_location" -> resources.getString(R.string.geocoded_location)
+                "last_reboot" -> resources.getString(R.string.last_reboot)
+                "light_sensor" -> resources.getString(R.string.light_sensor)
+                "location_background" -> resources.getString(R.string.pref_location_background_summary)
+                "next_alarm" -> resources.getString(R.string.next_alarm)
+                "phone_state" -> resources.getString(R.string.phone_state)
+                "pressure_sensor" -> resources.getString(R.string.pressure_sensor)
+                "proximity_sensor" -> resources.getString(R.string.proximity_sensor)
+                "sim_1" -> resources.getString(R.string.sim_1)
+                "sim_2" -> resources.getString(R.string.sim_2)
+                "steps_sensor" -> resources.getString(R.string.steps_sensor)
+                "storage_sensor" -> resources.getString(R.string.storage_sensor)
+                "wifi_connection" -> resources.getString(R.string.wifi_connection)
+                "zone_background" -> resources.getString(R.string.pref_location_zone_summary)
+                else -> resources.getString(R.string.no_description)
```

Comment:
Might want to prepend something to these keys so we know what they are, maybe `sensor_description_*`?

Reply:
Also it almost feels like this should be part of the `BasicSensor`.  That way people know to add a description when they add a sensor rather than having to come here too.  You could pass the basic sensor instead of just the `sensorId` in the constructor.

Reply:
Yea I honestly wasn't sure what the best approach would be here. After I got it working I figured send it over for review and see what can be improved upon.  I do agree having it as part of `BasicSensor` makes sense I'll have to see how to do that 😃

Reply:
I would just add the `R.string.sensor_description`, which should be an int, in the basic sensor.  Then in the preference just do `getString(sensor.description)`